### PR TITLE
py sol for Modern Art

### PR DESCRIPTION
**Pull request by freakin23**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [ ] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/silver/usaco-744.mdx [hunk 1]
```diff
@@ -208,4 +208,72 @@ public class Art {
...
+	for c in range(width):
+		color = canvas[r][c]
+		if color not in visible:
+			visible[color] = {"min_r": r, "max_r": r, "min_c": c, "max_c": c}
+		else:
+			bounds = visible[color]
+			bounds["min_r"] = min(r, bounds["min_r"])
+			bounds["max_r"] = max(r, bounds["max_r"])
+			bounds["min_c"] = min(c, bounds["min_c"])
+			bounds["max_c"] = max(c, bounds["max_c"])
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-744.mdx:229]: use a namedtuple or dataclass

Comment (freakin23) [reply solutions/silver/usaco-744.mdx]: ok

File: solutions/silver/usaco-744.mdx [hunk 2]
```diff
@@ -208,4 +208,74 @@ public class Art {
 ```
 
 </JavaSection>
+<PySection>
+
+```py
+class Bound:
+	def __init__(self, min_r: int = 0, max_r: int = 0, min_c: int = 0, max_c: int = 0):
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-744.mdx:215]: default argument not needed

File: solutions/silver/usaco-744.mdx [hunk 3]
```diff
@@ -208,4 +208,74 @@ public class Art {
...
+	width = int(read.readline().strip())
+	canvas = [list(map(int, read.readline().split())) for _ in range(width)]
+
+visible = {}
+for r in range(width):
+	for c in range(width):
+		color = canvas[r][c]
+		if color not in visible:
+			visible[color] = Bound(r, r, c, c)
+		else:
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-744.mdx:232]: else not needed

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/silver/usaco-744.mdx (hunk 2 above)
Comment (freakin23) [reply solutions/silver/usaco-744.mdx:215]: ok

File: solutions/silver/usaco-744.mdx (hunk 3 above)
Comment (freakin23) [reply solutions/silver/usaco-744.mdx:232]: ok

**Pull request merged**